addons/transifex/models/ir_translation.py

```diff
@@ -0,0 +1,78 @@
+from odoo.modules.module import ad_paths
+
+
+class IrTranslation(models.Model):
+
+    _inherit = 'ir.translation'
+
+    transifex_url = fields.Char("Transifex URL", compute='_get_transifex_url')
+
+    def _get_transifex_url(self):
+        """ Construct transifex URL based on the module on configuration """
+        # e.g. 'https://www.transifex.com/odoo/'
+        base_url = self.env['ir.config_parameter'].sudo().get_param('transifex.project_url')
+
+        tx_config_files = ""
+        for ad_path in ad_paths:
+            tx_path = opj(opj(ad_path, '.tx'), 'config')
+            if os.path.isfile(tx_path):
+                with open(tx_path, 'r') as tx_file:
+                    tx_config_files += tx_file.read()
+
+            # parent directory ad .tx/config is root directory in odoo/odoo
+            tx_path = opj(opj(opj(ad_path, os.pardir), '.tx'), 'config')
+            if os.path.isfile(tx_path):
+                with open(tx_path, 'r') as tx_file:
+                    tx_config_files += tx_file.read()
+
+        if not base_url or not tx_config_files:
+            self.transifex_url = False
+        else:
+            base_url = base_url.rstrip('/')
+
+            # will probably be the same for all terms, avoid multiple searches
+            translation_languages = list(set(self.mapped('lang')))
+            languages = self.env['res.lang'].with_context(active_test=False).search(
+                [('code', 'in', translation_languages)])
+            language_codes = dict((l.code, l.iso_code) for l in languages)
+
+            # .tx/config files contains the project referenche
+            # using ini files like '[odoo-master.website_sale]'
+            translation_modules = list(set(self.mapped('module')))
+            project_modules = {}
+            for module in translation_modules:
+                p = re.compile('\[([\w\-]+)\.%s\]' % module)
+                match = p.findall(tx_config_files)
+                if match:
+                    project_modules[module] = match[0]
+
+            for translation in self:
+                if not translation.module or not translation.source or translation.lang == 'en_US':
+                    # custom or source term
+                    continue
+
+                lang_code = language_codes.get(translation.lang)
+                if not lang_code:
+                    continue
+
+                project = project_modules.get(translation.module)
+                if not project:
+                    continue
+
+                # e.g. 'https://www.transifex.com/odoo/odoo-10/translate/#fr/sale/42?q=Sale+Order'
+                translation.transifex_url = "%(url)s/%(project)s/translate/#%(lang)s/%(module)s/42?q=%(src)s" % {
+                    'url': base_url,
+                    'project': project,
+                    'lang': lang_code,
+                    'module': translation.module,
+                    'src': werkzeug.url_quote_plus(translation.source[:50]),
```

Comment:
If two sources have the same leading 50 chars, does this lead to a screen showing both?

Reply:
yes

Reply:
😲